addons/package_hierarchy/models/stock_quant_package.py

```diff
@@ -271,3 +280,36 @@ def _get_package_link_structure(self):
             )
         hierarchy_lines = hierarchy_lines | package_hierarchy_lines
         return hierarchy_lines.ids
+
+    def get_or_construct_hierarchy_from_packages(self):
+        """Getting or Creating (if any of hierarchy nodes doesnt exist) all hierarchy links of
+        ordered packages that are in self recordset.
+
+        :return: recordset of package hierarchy links
+        """
+        # Setting self to sudo to over pass access rights at this moment, have to be reviewed later
+        self = self.sudo()
+        PackageHierarchyLink = self.env["package.hierarchy.link"]
+
+        # Empty recordset
+        package_hierarchies = self.env["package.hierarchy.link"]
```

Comment:
Shouldn't this be `PackageHierarchyLink.browse()` ?

Reply:
It is the same thing in fact, but yeah due to what have been discussed about coding standards happy to make that change